### DISCUSSION TITLE
Treating warnings as errors because process.env.CI = true.

### DISCUSSION THREAD

**Discussion by j-o-o** [vercel/community#30] [Help] [answered]
While deploying my Create React App, it gives me this error:

```
Treating warnings as errors because process.env.CI = true.
```

and my build fails. Why?

**Comment by saulofilho** [comment]
Hi. You need override the build command. Try this:  `CI='' yarn build` or  `CI='' npm run build`
:)

**Reply by tsarg88** [reply]
Was having a similar issue, thanks!

**Reply by primelos** [reply]
Hi, first time i see this error... I'm still learning where do I add CI='' npm run build

**Reply by Manvityagi** [reply]
@primelos In your .yml workflow file replace `npm run build` with `CI='' npm run build`

**Reply by MrJadaml** [reply]
I'm just testing out vercel, importing in a repo and in that interface there is an option to "override" your build command.

**Reply by RecepCil** [reply]
I had similar issue and this fixed mine, thanks!

**Reply by mayank1513** [reply]
Hi, where do you put this line `CI='' npm run build` in github action yml file?

**Reply by RecepCil** [reply]
> 
> 
> Hi, where do you put this line `CI='' npm run build` in github action yml file?

Hi, you can check from [this link](https://github.com/RecepCil/mtdb/commit/d95f09b64b080e30d0758ffa5f1c4713b362451e)

**Reply by ilyapasyuk** [reply]
Why not just `CI=false npm run build` command?

**Reply by harikanani** [reply]
The same Problem Solved, 
Thanks

**Reply by desdamo** [reply]
Hello sorry but i dont inox d'heure change something i think its here right?

```
"scripts": { 
"start": "craco start", 
"build": "CI='' yarn build", 
"test": "craco test", 
"eject": "react-scripts eject" },
```

Because i tried like this and it didn't work with vercel
Thanks

**Reply by mat3000** [reply]
@desdamo FYI: https://vercel.com/support/articles/how-do-i-resolve-a-process-env-ci-true-error?query=env.CI

**Reply by AndyHIT** [reply]
"scripts": {
    "start": "react-scripts start",
    "build": "CI=false react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  }

**Reply by jenoturkey** [reply]
I have no idea how to get rid of the clone someone I know has it on my xiaomi M10 phone and I don't know what to do to get just my phone back

**Reply by primelos** [reply]
so @huandrey solution worked for me.  @Manvityagi I don't even have a .yml do i create it (right click add file)? or do I run npm <???>

**Reply by mhmdAzzouzi** [reply]
Thank you, this was really helpful

**Reply by ibrokhimismoilov** [reply]
> Hi. You need override the build command. Try this: `CI='' yarn build` or `CI='' npm run build`
> :)

Thank you so much

**Reply by nick-tw** [reply]
Why doesn't Verel fill this in for me whenever I generate a react website?

Every time I start a new site via create-react-app on Vercel, I always have to google this issue.

**Reply by jeflucas** [reply]
Thanks!

**Reply by warengonzaga** [reply]
Thanks! This is very helpful!

**Reply by firdausbaihaqi** [reply]
Thank you!! but can someone explain what is happening behind the scene after we override our build command with this?

**Reply by JonathanYon** [reply]
is  `CI='' yarn build` the same as `Cl= yarn build` ?????

**Reply by zenggong965** [reply]
I just opened BUILD COMMAND OVERRIDE then paste CI='' npm run build, and it worked.
Thanks!

**Reply by goncanegis** [reply]
I come here ever time I build and deploy on Vercel. Thank you fellow internet user for this solution!

**Reply by TASLIM-ENIOLLA** [reply]
Where do I find the .yml workflow in my react project? All I can see is package & package.lock JSON files.

**Reply by leerob** [reply]
https://github.com/vercel/community/discussions/30#discussioncomment-2416205

**Comment by huandrey** [comment]
### steps 
- dashboard -> open your project -> settings -> Environment Variables -> follow the screenshot and save the variable.
- Redeploy your project

![image](https://user-images.githubusercontent.com/62030386/113489409-85776180-949a-11eb-8511-51f919aa4e61.png)

**Reply by soheilous** [reply]
its worked for me . thanks .

**Reply by kyotodevIndie** [reply]
worked for me too, ty bro

**Reply by burfbari** [reply]
thanks for the Vercel steps.

**Reply by trangiabao-hub** [reply]
really appreciate, I work for me, thank you guy

**Reply by GeraCV** [reply]
Por fin encuentro la solución, muchas gracias.

**Reply by KNehe** [reply]
Thanks. Worked for me too

**Reply by primelos** [reply]
worked for me! thanks

**Reply by NevetsKuro** [reply]
adding a new Environment Variable worked for me too! Thanx. 👍

**Reply by gsambrotta** [reply]
Doesn't this mean doesn't use CI for deploying anymore?

**Reply by onataliaduran** [reply]
Thanks! it worked for me 👍

**Reply by Oliveiras96** [reply]
It worked for  me, thanks!

**Reply by jameserie** [reply]
Thank you

**Reply by raghavddps2** [reply]
This worked for me too. Thank you :)

**Reply by Hermogenes00** [reply]
Thank you

**Reply by Chalithya** [reply]
It worked for me. Thank you!

**Reply by Denver44** [reply]
Thank you, It worked for me

**Reply by sreashi** [reply]
Thank you after trying for an hour it finally works...thank you

**Reply by RafaelFigueiredo2203** [reply]
Funcionou pra mim!

**Reply by libio** [reply]
Thank you.

**Reply by GabrielSantosLemos** [reply]
Show cara obrigadão.

**Reply by victorbohrer** [reply]
worked for me, thanks!!

**Reply by lucas756** [reply]
thank you.

**Reply by Prathamesh-Shanbhag** [reply]
Worked for me as well, Thank you so much!

**Reply by IhorMorozov** [reply]
Thank you!!!

**Reply by lcds90** [reply]
Thanks!

**Reply by ghulamyazdani** [reply]
Worked for me!

**Reply by pvtdao** [reply]
It worked!

**Reply by TheYass1n** [reply]
It work

**Reply by db-Square** [reply]
works for me

**Reply by jblici** [reply]
THANK YOU

**Reply by matheusmota-dev-br** [reply]
Thanks!

**Reply by JoydipGhoshAcademy** [reply]
Oh holly sh*** It works perfectly for me!!!!!. Amazing I want to donate some bitcoin. Please let me know how to donate bro

**Reply by huandrey** [reply]
haha, nice bro!! I'm glad to have helped you :))
my discord contact is this Huandrey#6906

**Reply by JuFortini** [reply]
It did work!! Thank you so much, I was trapped in this.

**Reply by rohjs** [reply]
Thanks for the Vercel steps! It really helped 👍

**Reply by JoydipGhoshAcademy** [reply]
give ur btc/solana address!

**Reply by WahabShah23** [reply]
Naaaaiiiiccce!

**Reply by DiestroCorleone** [reply]
Worked as a charm. Thanks a lot!

**Reply by OsDroidi** [reply]
Thanks you saved my life

**Reply by tohidshabanloo** [reply]
Thank you , finally worked.

**Comment by FarhanMuftiHilmy** [comment]
You can go to your project settings, choose general and below Build & Development Settings click override build command toggle button, input `CI='' npm run build`, save it, then redeploy your project. It should work fine.

**Reply by Newtdev** [reply]
Thanks. The worked!

**Reply by penguinuux** [reply]
Thanks. Really works.

**Comment by krishnadevz** [comment]
i did like this dashboard -> open your project -> settings -> Environment Variables -> then CI='' yarn build or CI='' npm run build it worked :)

**Comment by sreenathkspanikker** [comment]
@huandrey Thank you man its works.

**Comment by BelleNerissa** [comment]
Awesome tips!! Thanks all!

**Comment by leerob** [comment, marked as answer]
We now no longer fail CRA builds on Vercel that have warnings: https://github.com/vercel/vercel/pull/7584

Thank you for your feedback here 🙏

**Reply by kainosnoema** [reply]
We just had this bite us. One of our assets became too large to cache via service worker and CRA output a warning that should have failed in testing. CircleCI caught it and failed the build (as expected when CI=true, which is what you'd expect in CI), Vercel didn't and the failed caching caused page load errors after deployment.

Can we still explicitly set `CI=true` to revert to the old behavior?

**Reply by leerob** [reply]
Yep, you can still add `CI=true` to get the old behavior 👍
